### DISCUSSION TITLE
"error can't alloc pcb" occurs when calling fetch API.

### DISCUSSION THREAD

**Discussion by yuichisuzuki0601** [microsoft/devicescript#660] [Q&A] [answered]
* The device is Raspberry Pi pico.
* I have confirmed that the device is connected to wifi.
* It happens when I make an http request.
* Timeout occurs when making https request.

# error log

```
DEV> connecting to tcp://192.168.0.8:8080
DEV> socket tcp://192.168.0.8:8080 error can't alloc pcb
DEV> Unhandled exception
DEV> Exception: Error
DEV>  message: socket tcp://192.168.0.8:8080: can't alloc pcb
DEV>  at _connect [as F38_pc214] (node_modules/@devicescript/net/src/sockets.ts:158:29)
DEV>  at connect [as F20_pc7] (node_modules/@devicescript/net/src/sockets.ts:172:5)
DEV>  at fetch [as F15_pc534] (node_modules/@devicescript/net/src/fetch.ts:174:5)
DEV>  at inline [as F8_pc23] (src/main.ts:47:5)
DEV>  at emit [as F64_pc24] (node_modules/@devicescript/core/src/events.ts:23:17)
DEV> Unhandled exception
DEV> stop program
DEV> connection error: "sock write error"
DEV> connection error: "sock write error"
DEV> connection error: "sock write error"
DEV> connection error: "sock write error"
DEV> connection error: "sock write error"
```

I want to know the cause and solution.

**Comment by mmoskal** [comment, marked as answer]
I guess the error message could be better, but TLS is not supported with DeviceScript on Pico-W, please see https://microsoft.github.io/devicescript/devices#implementation-status

PRs are welcome!

**Reply by yuichisuzuki0601** [reply]
Thank you for your reply.

It seems that pico does not support TLS.
Even now, the same error also seems to occur after the second time when using TCP (http) communication.

If I have enough time to make a PR, I'll look into it!
However, I will wait for the new version of DeviceScript for a while!

regards.
